src/object/notification.c

```diff
@@ -103,6 +101,16 @@ void sendSignal(notification_t *ntfnPtr, word_t badge)
                     MCS_DO_IF_SC(tcb, ntfnPtr, {
                         possibleSwitchTo(tcb);
                     })
+#ifdef CONFIG_KERNEL_MCS
+                    if (tcb->tcbSchedContext != NULL && sc_active(tcb->tcbSchedContext)) {
+                        sched_context_t *sc = SC_PTR(notification_ptr_get_ntfnSchedContext(ntfnPtr));
+                        if (tcb->tcbSchedContext == sc) {
+                            /* Only unblock if the SC was donated from the
```

Comment:
If the thread receiving the signal is currently in the ThreadState_RunningVM state then doesn't it have to already have an SC? So there wouldn't be any donated SC in this case?

Reply:
I don't think there is any guarantee for that. It's possible that the thread was running, had a bound SC removed and then this was signalled such that the SC from the bound notification was then bound to the VM.

Reply:
I think we need an isSporadic check here too anyway